src/First/README.md

```diff
@@ -0,0 +1,451 @@
+
+concat(c, concat(b, a))
+//=> First( Just "a" )
+
+concat(concat(c, b), a)
+//=> First( Just "a" )
+
+concat(concat(a, b), c)
+//=> First( Just "c" )
+```
+
+#### option
+
+```haskell
+First a ~> a -> a
+```
+
+`First` wraps an underlying `Maybe` which provides the ability to option out
+a value in the case as of an empty instance. Just like `option` on a `Maybe`
+instance, it takes a value as its argument. When run on an empty instance,
+the provided default will be returned. If `option` is run on a non-empty
+instance however, the wrapped value will be extracted not only from the
+`Last`, but also from the underlying `Just`.
+
+If the underlying `Maybe` is desired, the [`valueOf`](#valueof) method can be
+used and will return the `Maybe` instead.
+
+```javascript
+const First = require('crocks/First')
+
+const compose = require('crocks/helpers/compose')
+const chain = require('crocks/pointfree/chain')
+const isString = require('crocks/predicates/isString')
+const mconcatMap = require('crocks/helpers/mconcatMap')
+const prop = require('crocks/Maybe/prop')
+const safe = require('crocks/Maybe/safe')
+
+// stringVal :: a -> Maybe String
+const stringVal = compose(
+  chain(safe(isString)),
+  prop('val')
+)
+
+// firstValid :: [ a ] -> First String
+const firstValid =
+  mconcatMap(First, stringVal)
+
+// good :: [ Object ]
+const good =
+  [ { val: 23 }, { val: 'string' }, { val: '23' } ]
+
+// bad :: [ Object ]
+const bad =
+  [ { val: 23 }, { val: null }, {} ]
+
+firstValid(good)
+  .option('')
+//=> "string"
+
+firstValid(bad)
+  .option('')
+//=> ""
+```
+
+#### valueOf
+
+```haskell
+First a ~> () -> Maybe a
+```
+
+`valueOf` is used on all `crocks` `Monoid`s as a means of extraction. While the
+extraction is available, types that implement `valueOf` are not necessarily a
+`Comonad`. This function is used primarily for convenience for some of the
+helper functions that ship with `crocks`. Calling `valueOf` on
+a `First` instance will result in the underlying `Maybe`.
+
+```javascript
+const First = require('crocks/First')
+
+const { Nothing } = require('crocks/Maybe')
+const valueOf = require('crocks/pointfree/valueOf')
+
+valueOf(First(56))
+//=> Just 56
+
+valueOf(First.empty())
+//=> Nothing
+
+First(37)
+  .concat(First(99))
+  .valueOf()
+//=> Just 37
+
+First(Nothing())
+  .concat(First.empty())
+  .valueOf()
+//=> Nothing
+```
+
+## Transformation Functions
+
+#### eitherToFirst
+
+`crocks/First/eitherToFirst`
+
+```haskell
+eitherToFirst :: Either b a -> First a
+eitherToFirst :: (a -> Either c b) -> a -> First b
+```
+
+Used to transform a given `Either` instance to a `First`
+instance, `eitherToFirst` will turn a `Right` instance into a non-empty `First`
+wrapping the original value contained in the `Right`. All `Left` instances will
+map to an empty `First`, mapping the originally contained value to a `Unit`.
+Values on the `Left` will be lost and as such this transformation is considered
+lossy in that regard.
```

Comment:
Is `lossy` an understood term? Should you be linking to an article or definition or something? Never heard this term in this context, myself, but I'm a noob...

Reply:
I use the term from Information Theory for source coding, on some encodings, for things like say a bitmap to a jpeg. I do not have an isomorphism from bitmap -> jpeg and jpeg -> bitmap and have them point back to the same exact bitmap on the round trip.

Same thing happens here, because `Either` has both `Left` and `Right` and the target `First` maps all `Lefts` to `Unit`, I can never get back to the same `Left` as I started with, without providing some value to map the `Unit` back into the original `Left`.

So I do not know if it is an equivalent term, but could not come up with one that is quite so close.

Do you have any suggestions on how I could make this more clear to the curious?

Reply:
A link to something on the subject is always a nice thing to have.

Reply:
Since this occurs any time we have a transformation with most sum types to `Maybe`, it will end up being a common theme. 
So I think that primer I mentioned a couple times could have a section on natural transformations with Sum Types.

But in the meantime, how would you feel about emphasizing this point in the code example with a comment over [this line](https://github.com/evilsoft/crocks/pull/195/files#diff-dfe56abb2d91236bdea5f40bdcb447d7R260) to the effect of:
```js
// 'Bad Times' is lost
eitherToFirst(Left('Bad Times'))
//=> First( Nothing )
```

Will also do the same with `resultToFirst`.
Do you think that may help?